### PR TITLE
[ENH] allow Levels values to be objects

### LINKED ISSUE #1573: allow Levels values (in sidecar for TSV files) to be objects
### Your idea

was raised during a meeting at OHBM that some project invovled with phenotypic annotation ([neurobagel](https://neurobagel.org/)) may greatly benefit if the data type for [TermURL](https://bids-specification.readthedocs.io/en/stable/glossary.html#objects.metadata.TermURL) (in the sidecar json of TSV files) could be a json object instead of just string. 

Opening an issue here to to start the conversation.

@ericearl @jbpoline @nikhil153 @surchs @michellewang @bcmcpher

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes #1573

### CODE REVIEW

File: src/common-principles.md [hunk 1]
```diff
@@ -513,6 +513,28 @@ Example:
...
+like in the example below.
+
+```JSON
+{
+    "sex": {
+        "Description": "sex of the participant as reported by the participant",
+        "Levels": {
+            "M": {
+                "Description":"male",
+                "TermURL": "TODO"
```
Comment (Remi-Gau) [inline src/common-principles.md:527]: update

File: src/common-principles.md [hunk 2]
```diff
@@ -513,6 +513,28 @@ Example:
...
+            },
+            "F": {
+                "Description":"female",
+                "TermURL": "TODO"
```
Comment (Remi-Gau) [inline src/common-principles.md:531]: update

**Comment by Remi-Gau** [comment]
@surchs @ericearl @alyssadai

**Comment by Remi-Gau** [comment]
@effigies am I correct that this is one aspect of the spec that has not been schemaified yet?

**Comment by Remi-Gau** [comment]
Should the examples of the json sidecar [for participants.tsv](https://bids-specification.readthedocs.io/en/latest/modality-agnostic-files.html#participants-file) and for [phenotypic assessments](https://bids-specification.readthedocs.io/en/latest/modality-agnostic-files.html#phenotypic-and-assessment-data) be updated to also include this way of describing levels?

File: src/common-principles.md [hunk 3]
```diff
@@ -513,6 +513,28 @@ Example:
 }
 ```
 
+Each level can be described with a string as in the example above,
+or with an object containing the fields `Description` and `TermURL`
```
Comment (surchs) [inline src/common-principles.md:517]: Do we expect BIDS users to know what `TermURL` expects? How about:

```suggestion
or with an object containing the fields `Description` and `TermURL`.
If you use the object format, the value of the `TermURL` key should
be a URL to a term from a controlled vocabulary that describes the
level. For example: https://www.ncbi.nlm.nih.gov/mesh/68008297 for 
"male". The value of the `Description` key should then correspond
to the human readable name of this term (or to the preferred name
if several options exist in the controlled vocabulary). For example: "Male".
```

File: src/common-principles.md (hunk 1 above)
Comment (surchs) [inline src/common-principles.md:527]:
```suggestion
                "Description": "Male",
                "TermURL": "https://www.ncbi.nlm.nih.gov/mesh/68008297"
```

File: src/common-principles.md (hunk 2 above)
Comment (surchs) [inline src/common-principles.md:531]:
```suggestion
                "Description": "Female",
                "TermURL": "https://www.ncbi.nlm.nih.gov/mesh/68005260"
```

**Comment by effigies** [comment]
> @effigies am I correct that this is one aspect of the spec that has not been schemaified yet?

Correct. It's a weird state because what we really need to say is that, if the suffix is TSV, then additional properties of sidecars have the form of an object with certain allowed terms. It would have been much more convenient if we had made the column descriptions an object in a single field. (Further, there generally aren't rules against extra metadata, so it's actually that we need to create a typed metadata field dynamically based on the headers in the TSV.)

File: src/common-principles.md (hunk 3 above)
Comment (effigies) [reply src/common-principles.md:517]: If we're going to improve the definition of TermURL, we should do it in https://bids-specification.readthedocs.io/en/stable/glossary.html#termurl-metadata so it appears in the glossary and any tables it's defined in.

 https://bids-specification.readthedocs.io/en/stable/glossary.html#termurl-metadata

Comment (Remi-Gau) [reply src/common-principles.md]: I could only transfer one sentence to the TermURL generic definition, the rest of @surchs suggestion feels too specific to this example I think.

**Review by effigies — APPROVED** [review]

**Review by surchs — APPROVED** [review]
:tada: thanks a lot @Remi-Gau. One free bagel on us!

**Review by sappelhoff — APPROVED** [review]
LGTM! Thanks a lot @Remi-Gau

**Pull request merged**